app/controllers/admin/users_controller.rb

```diff
@@ -21,5 +23,61 @@ def unlock_account
       redirect_to user_search_path, notice: t("admins.user_search.account_unlocking_scheduled", name: u.username)
     end
 
+    def make_admin
+      u = User.find(params[:id])
+      Role.add_admin u.person
+      redirect_to user_search_path, notice: t("admins.user_search.add_admin", name: u.username)
+    end
+
+    def remove_admin
+      u = User.find(params[:id])
+      Role.remove_admin u.person
+      redirect_to user_search_path, notice: t("admins.user_search.delete_admin", name: u.username)
+    end
+
+    def make_moderator
+      u = User.find(params[:id])
+      Role.add_moderator u.person
+      redirect_to user_search_path, notice: t("admins.user_search.add_moderator", name: u.username)
+    end
+
+    def remove_moderator
+      u = User.find(params[:id])
+      Role.remove_moderator u.person
+      redirect_to user_search_path, notice: t("admins.user_search.delete_moderator", name: u.username)
+    end
+
+    def make_spotlight
+      u = User.find(params[:id])
+      Role.add_spotlight u.person
+      redirect_to user_search_path, notice: t("admins.user_search.add_spotlight", name: u.username)
+    end
+
+    def remove_spotlight
+      u = User.find(params[:id])
+      Role.remove_spotlight u.person
+      redirect_to user_search_path, notice: t("admins.user_search.delete_spotlight", name: u.username)
+    end
+
+    private
+
+    def validate_user
+      redirect_to user_search_path, notice: t("admins.user_search.does_not_exist") unless User.exists?(params[:id])
+    end
```

Comment:
How about something like 
```rb
@user = User.where(id: params[:id]).first
redirect_to user_search_path, notice: t("admins.user_search.does_not_exist") unless @user
```

And using `@user` in the actions instead of doing another query?